definitions/npm/react-apollo_v2.x.x/flow_v0.58.x-/react-apollo_v2.x.x.js

```diff
@@ -276,16 +276,16 @@ declare module 'react-apollo' {
 
   declare export class Subscription<TData> extends React$Component<SubscriptionProps<TData>> {}
 
-  declare export type MutationFunction<TVariables> = (options: {
+  declare export type MutationFunction<TData=void, TVariables=void> = (options: {
     variables?: TVariables,
     optimisticResponse?: Object,
     refetchQueries?: (mutationResult: FetchResult) => Array<{query: DocumentNode, variables: {[string]: any}}>,
     update?: (cache: DataProxy, mutationResult: FetchResult) => any
-  }) => Promise<*>
+  }) => Promise<MutationResult<TData>>
```

Comment:
This doesn't seem correct. Types here have it as `Promise<void | FetchResult<TData>>`
https://github.com/apollographql/react-apollo/blob/master/src/Mutation.tsx#L52